packages/l/libsdl_image/xmake.lua

```diff
@@ -43,19 +52,15 @@ package("libsdl_image")
 
     on_install("macosx", "linux", function (package)
         local configs = {}
-        if package:config("shared") then
-            table.insert(configs, "--enable-shared=yes")
-        else
-            table.insert(configs, "--enable-shared=no")
-        end
-        if package:is_plat("linux") and package:config("pic") ~= false then
-            table.insert(configs, "--with-pic")
-        end
+        table.insert(configs, "--enable-shared=" .. (package:config("shared") and "yes" or "no"))
+        table.insert(configs, "--enable-static=" .. (package:config("shared") and "no" or "yes"))
         local libsdl = package:dep("libsdl")
         if libsdl and not libsdl:is_system() then
             table.insert(configs, "--with-sdl-prefix=" .. libsdl:installdir())
+        elseif libsdl and package:is_plat("macosx") then
+            table.insert(configs, "--with-sdl-prefix=" .. libsdl:installdir())
```

Comment:
though that didn't work before

Reply:
hm, weird that it still can't find the method

Reply:
especially since it says `building for macOS-x86_64 but attempting to link with file built for macOS-x86_64`

Reply:
After I reinstalled libsdl and use 2.24.0, I was able to reproduce it locally, but I don't know how to fix it.

It works if it use libsdl 2.0.22, but it does not work for libsdl 2.24.0

Reply:
> especially since it says `building for macOS-x86_64 but attempting to link with file built for macOS-x86_64`

then libSDL2_image.a will be ignore when doing link.

Reply:
It should be configure's bug. I have used cmake to fix it, but other libsdl project need use cmake too.